### PR TITLE
Using PreallocationTools.jl

### PR DESCRIPTION
**Pull request by jpfairbanks**
@ChrisRackauckas, I added some fixes for #171

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Awesome, yeah there was just a few dangling things there 😅. Examples of calibration to come when I get in-flight wifi up.

**Comment by ChrisRackauckas** [comment]
This looks good to me.

File: src/simulation.jl
Comment (lukem12345) [inline src/simulation.jl:77]: PR #151 also makes this refactoring and pulls it into a function to be called elsewhere, but I will leave this be and fix the merge conflict when it is time to merge 151.

**Review by lukem12345 — COMMENTED** [review]
Changes look good to me. Just reran the CISM benchmark on this branch on HPG, and times are sitting at just over 11 [ms] using the Tsit5 solver. Loading up CairoMakie now to visually inspect results as final sanity check, though RMS and max difference appear normal. Will approve pending the render.

**Comment by lukem12345** [comment]
I should note that I was unable to get the [parallel solver](https://docs.sciml.ai/DiffEqDocs/stable/solvers/ode_solve/#Parallel-Explicit-Runge-Kutta-Methods) `KuttaPRK2p5` to work due to `ERROR: `@threads :static` cannot be used concurrently or nested`.

**Comment by lukem12345** [comment]
Looks like artifiacting builds up in strange ways using the FDF5 solver. Tsit5 results are included for comparison. This is very likely not a "bug" in the solver implementation nor in Decapodes per se. But just thought that it should be noted.

Let me know your thoughts @jpfairbanks and I will go ahead and merge.

fdf5
tsit5

**Review by lukem12345 — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
That goes away at lower tolerances?

**Comment by lukem12345** [comment]
Presumably yeah. That's been my experience using the explicit timestepping solvers with Decapdoes. I'll run some more tests with FBDF this afternoon and open an issue if changing up the parameters doesn't resolve this. The GIFs here are using just the default parameters.

**Comment by jpfairbanks** [comment]
The artifacts shouldn't hold up the merge. We can diagnose the numerics in a separate issue.

**Comment by ChrisRackauckas** [comment]
> I should note that I was unable to get the [parallel solver](https://docs.sciml.ai/DiffEqDocs/stable/solvers/ode_solve/#Parallel-Explicit-Runge-Kutta-Methods) KuttaPRK2p5 to work due to ERROR: @threads :static cannot be used concurrently or nested. But since the handful of implicit solvers that I tried work, this shouldn't block this PR.

That one probably won't be a good idea for this problem.

**Pull request merged**
